src/figuresClasses.ts

```diff
@@ -1,19 +1,91 @@
-export interface Figure {
+export enum Shape {
+  Triangle = 'triangle',
+  Circle= 'circle',
+  Rectangle = 'rectangle',
+}
 
+export enum Color {
+  Red = 'red',
+  Blue = 'blue',
+  Green = 'green',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape,
+  color: Color,
 
+  getArea(): number,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || b <= 0) {
+      throw new Error('The side length cannot be less than or equal to 0');
+    }
```

Comment:
You've used same logic in every class, so would be nice to move it to helper function